src/marten/db/query/sql/query.cr

```diff
@@ -74,6 +74,25 @@ module Marten
             ensure_join_for_field_path(field_path, selected: true)
           end
 
+          def average(raw_field : String)
+            field_context = get_field_context(raw_field, Model)
+
+            unless field_context.field.is_a?(Marten::DB::Field::Int) ||
+                   field_context.field.is_a?(Marten::DB::Field::Float)
+              raise Errors::InvalidField.new(
+                "Cant calculate the average of '#{raw_field}' (#{field_context.field.class})."
+              )
+            end
```

Comment:
I think here we can avoid hardcoding rules regarding what fields are targeted because technically calculating averages should work for `BigInt` fields too for example. Moreover, some custom fields could also correspond to numeric fields at the database level without inheriting from built-in fields like `Int` or `Float`. Calling `#average` should work for those ideally I think.

In this situation, it's probably safer to avoid this check and let the underlying DB driver return an error if the targeted column cannot be averaged. What do you think?